### ISSUE TITLE
Django settings.py

### ISSUE THREAD

**Issue by kingoftherug** [cannatag/ldap3#360]
Hello Giovanni.
at 1st thank you very much for cool ldap solution and detailed documentations it's very useful.
I'm beginner in python programming so my question can be so easy for you but doesn't for me.
I can't find any examples for using ldap3 with django and doesn't understand how can it be configured into settings.py. Could you explain for me or add to documentation some few examples abount use ldap3+django. 
best regard.
Roman.

**Comment by cannatag** [comment]
Hi,
to use ldap3 with django you can try the [django-python3-ldap](https://github.com/etianen/django-python3-ldap). I don't use django so I can't help you with some examples.

Bye,
Giovanni

**Comment by jcarswell** [comment]
Hi @kingoftherug

Provided you are building a custom authentication module and having issues configuring the variables, the best thing to do would be local settings file of built in as part of the main python application where you are preforming the actual bind. This would be the best option if you plan on making you module portable. If you are not wanting to go that route but have it as part of the main config you can import the settings using _from django.conf import settings_ and use your global defined variables that way.

I am assuming that is what you are looking at doing. If you are looking at extending the built in authentication I would recommend reading up on [writing an authentication backend](https://docs.djangoproject.com/en/1.11/topics/auth/customizing/#writing-an-authentication-backend).
Basically you would import ldap3 here and implement the required function get_user(user_id) and authenticate(request, **credentials). Alternatively you could look at adding the pre-built module [djanjo-auth-ldap] (https://pypi.python.org/pypi/django-auth-ldap/1.2.8) module, which unfortunately doesn't utilize this Great library. 

Hope that helps you out

**Comment by kingoftherug** [comment]
Hi @jcarswell 
Thank you for not being indifferent to my question.
My task is create modul which can get/update users into my db from AD by ldap. At this stage I don't need to make authentication of my programm. just all I need is connect to AD by one user(service user) which get all necessary data for me and I don't want keep connection data (user,pwd,server) in script. I already done such script with ldap3 solution. I placed connection parameters into settings.py as constants of my django project. I declared them as dictionary like
ldap3_setup = {'user' : <value>,
'password' : <value>,
'server' : <value>}

but
I have doubts about the correctness of this decision because everytime I should do impot of the dictionary from settings.py of my project.

**Comment by jcarswell** [comment]
@kingoftherug 
I don't necessarily think that is the write or wrong way of doing things, as there are several ways of storing that information. seeing different solutions around, they range being part of the script, in a separate config file within the module, part of the main configuration file, or as part of the model. your implementation will depend on how flexible and/or portable you want the solution.

The only downside to storing it in the main settings.py is that when moving the module from your dev environment to your production environment or someone else imports the module, they/you will need to add the necessary to the settings.py.
